test/e2e/resize_nodes.go

```diff
@@ -230,109 +270,213 @@ var _ = Describe("ResizeNodes", func() {
+					testName, supportedProviders, testContext.Provider))
+				return
+			}
+
+			if testContext.CloudConfig.NumNodes < 2 {
+				By(fmt.Sprintf("skipping %s test, which requires at lease 2 nodes (not %d)",
+					testName, testContext.CloudConfig.NumNodes))
+				return
+			}
+
+			// Create a replication controller for a service that serves its hostname.
+			// The source for the Docker containter kubernetes/serve_hostname is in contrib/for-demos/serve_hostname
+			name := "my-hostname-net-" + string(util.NewUUID())
+			createServiceWithNameSelector(c, ns, name)
+			replicas := testContext.CloudConfig.NumNodes
+			createServeHostnameReplicationController(c, ns, name, replicas)
+			pods, err := waitForPodsCreated(c, ns, name, replicas)
+			Expect(err).NotTo(HaveOccurred())
+			e := waitForPodsRunning(c, pods)
+			if len(e) > 0 {
+				Failf("Failed to wait for pods running: %v", e)
+			}
+			err = verifyPodsResponding(c, ns, name, pods)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("cause network partition on one node")
+			nodelist, err := c.Nodes().List(labels.Everything(), fields.Everything())
+			Expect(err).NotTo(HaveOccurred())
+			node := nodelist.Items[0]
+			pod, err := waitForPodOnNode(c, ns, name, node.Name)
+			Expect(err).NotTo(HaveOccurred())
+			Logf("Getting external IP address for %s", name)
+			host := ""
+			for _, a := range node.Status.Addresses {
+				if a.Type == api.NodeExternalIP {
+					host = a.Address + ":22"
+					break
+				}
+			}
+			Logf("Setting network partition on %s", node.Name)
+			dropCmd := fmt.Sprintf("sudo iptables -I OUTPUT 1 -d %s -j DROP", testContext.CloudConfig.MasterName)
+			if _, _, code, err := SSH(dropCmd, host, testContext.Provider); code != 0 || err != nil {
+				Failf("Expected 0 exit code and nil error when running %s on %s, got %d and %v",
+					dropCmd, node, code, err)
+			}
+
+			Logf("Waiting for node %s to be not ready", node.Name)
+			waitForNodeToBe(c, node.Name, false, 2*time.Minute)
+
+			Logf("Waiting for pod %s to be removed", pod.Name)
+			waitForPodToDisappear(c, ns, name, pod.Name)
+
+			By("verifying whether the pod from the partitioned node is recreated")
+			pods, err = waitForPodsCreated(c, ns, name, replicas)
+			Expect(err).NotTo(HaveOccurred())
+			e = waitForPodsRunning(c, pods)
+			if len(e) > 0 {
+				Failf("Failed to wait for pods running: %v", e)
+			}
+			err = verifyPodsResponding(c, ns, name, pods)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("remove network partition")
+			undropCmd := "sudo iptables --delete OUTPUT 1"
+			if _, _, code, err := SSH(undropCmd, host, testContext.Provider); code != 0 || err != nil {
+				Failf("Expected 0 exit code and nil error when running %s on %s, got %d and %v",
+					undropCmd, node, code, err)
+			}
+
+			Logf("Waiting for node %s to be ready", node.Name)
+			waitForNodeToBe(c, node.Name, true, 2*time.Minute)
+
+			By("verify wheter new pods can be created on the re-attached node")
```

Comment:
I've factored-out method waitForPodsCreatedRunningResponding. The code looks much better now.